runtime/compiler/runtime/ClassUnloadAssumption.cpp

```diff
@@ -1236,9 +1241,16 @@ J9::PersistentInfo::isUnloadedClass(
       void *v,
       bool yesIReallyDontCareAboutHCR)
    {
-   OMR::CriticalSection isUnloadedClass(assumptionTableMutex);
-   bool result = (_unloadedClassAddresses && _unloadedClassAddresses->mayContain((uintptrj_t)v));
-   return result;
+   if (getJITaaSMode() != SERVER_MODE)
+      {
+      OMR::CriticalSection isUnloadedClass(assumptionTableMutex);
+      return _unloadedClassAddresses && _unloadedClassAddresses->mayContain((uintptrj_t)v);
+      }
+   else
+      {
+      OMR::CriticalSection isUnloadedClass(TR::compInfoPT->getClientData()->getROMMapMonitor());
```

Comment:
TR::compInfoPT uses thread local storage which is more expensive to access and here we call it twice in a row. I would prefer to cache the clientSession to a local and use only one TLS access.